influxdb/src/main/java/com/hazelcast/jet/influxdb/InfluxDbSources.java

```diff
@@ -0,0 +1,260 @@
+     * @param username username of the InfluxDb server
+     * @param password password of the InfluxDb server
+     * @param pojoClass    the POJO class instance
+     * @param <T>      the POJO class
+     * @return a source to use in {@link com.hazelcast.jet.pipeline.Pipeline#drawFrom}
+     */
+    @Nonnull
+    public static <T> BatchSource<T> influxDb(@Nonnull String query, @Nonnull String database, @Nonnull String url,
+                                              @Nonnull String username, @Nonnull String password,
+                                              @Nonnull Class<T> pojoClass) {
+        checkNotNull(query, "query cannot be null");
+        checkNotNull(url, "url cannot be null");
+        checkNotNull(database, "database cannot be null");
+        checkNotNull(username, "username cannot be null");
+        checkNotNull(password, "password cannot be null");
+        checkNotNull(pojoClass, "pojoClass cannot be null");
+
+        return influxDb(query, () -> connect(url, username, password).setDatabase(database), pojoClass);
+    }
+
+    /**
+     * Creates a source that connects to InfluxDB database using the given
+     * connection supplier and emits items which are mapped to the provided
+     * POJO class type.
+     *
+     * @param query              query to execute on InfluxDb database
+     * @param connectionSupplier supplier which returns {@link InfluxDB} instance
+     * @param pojoClass    the POJO class instance
+     * @param <T>      the POJO class
+     * @return a source to use in {@link com.hazelcast.jet.pipeline.Pipeline#drawFrom}
+     */
+    @Nonnull
+    public static <T> BatchSource<T> influxDb(@Nonnull String query,
+                                              @Nonnull SupplierEx<InfluxDB> connectionSupplier,
+                                              @Nonnull Class<T> pojoClass) {
+        checkNotNull(query, "query cannot be null");
+        checkNotNull(connectionSupplier, "username cannot be null");
+        checkNotNull(pojoClass, "pojoClass cannot be null");
+
+        return SourceBuilder.batch("influxdb",
+                ignored -> new InfluxDbSourceContext<>(query, connectionSupplier, pojoClass, null))
+                .<T>fillBufferFn(InfluxDbSourceContext::fillBufferWithPojoMapping)
+                .destroyFn(InfluxDbSourceContext::close)
+                .build();
+    }
+
+    /**
+     * A source context around the InfluxDB connection that executes a query on
+     * InfluxDb and streams the results.
+     *
+     * @param <T> emitted item type
+     */
+    private static class InfluxDbSourceContext<T> {
+
+        /**
+         * Default number of {@link QueryResult}s to process in one chunk
+         */
+        private static final int DEFAULT_CHUNK_SIZE = 100;
+        private static final int MAX_FILL_ELEMENTS = 100;
+
+        private final Class<T> pojoClass;
+        private final BlockingQueue<QueryResult> queue = new ArrayBlockingQueue<>(10000);
```

Comment:
10000 is quite a lot. One result can contain 100 items, that's 1M results in total. Did this number come from some testing?